client/network/src/protocol.rs

```diff
@@ -785,25 +774,13 @@ where
 				}],
 			},
 		);
+		self.chain_sync.process_block_response_data(blocks_to_import);
```

Comment:
This makes no sense, above we call to sync to process the block data and then we call here again to process the response? Why isn't this done directly in the first call?